### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by RDPerera**
## Purpose

- Update the README files with accurate and working code examples.
- Add changelog.md.
- Update CLI documentation.

Resolves https://github.com/ballerina-platform/ballerina-library/issues/5949, Resolves https://github.com/ballerina-platform/ballerina-library/issues/5974

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: edi-cli/src/main/resources/cli-docs/edi.help [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+SYNOPSIS
+       bal edi <command> [OPTIONS] [<arguments>]
+
+DESCRIPTION
+       The 'bal edi' command-line tool provides the functionality required to process
+       EDI files and implement EDI integrations.
+
+COMMANDS
+       codegen
+           Generate records and parsing functions for a given EDI schema.
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply edi-cli/src/main/resources/cli-docs/edi.help]: Changed with https://github.com/ballerina-platform/edi-tools/pull/29/commits/3c80fe81036dbc049bc518828fc5f8e8aa24d079.

File: edi-cli/src/main/resources/cli-docs/codegen.help [hunk 2]
```diff
@@ -0,0 +1,20 @@
+NAME
+       bal edi codegen - Ballerina EDI Code Generation
+
+SYNOPSIS
+       bal edi codegen [OPTIONS]
+
+DESCRIPTION
+       The 'bal edi codegen' command is used to generate records and 
+       parsing functions for a given EDI schema.
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply edi-cli/src/main/resources/cli-docs/codegen.help]: Changed with https://github.com/ballerina-platform/edi-tools/commit/3c80fe81036dbc049bc518828fc5f8e8aa24d079.

File: edi-cli/src/main/java/io/ballerina/edi/cmd/CodegenCmd.java [hunk 3]
```diff
@@ -21,26 +21,27 @@
...
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 
-@CommandLine.Command(
-        name = "codegen",
-        description = "Generated code for a given EDI schema."
-)
+@CommandLine.Command(name = "codegen", description = "Generated code for a given EDI schema.")
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply edi-cli/src/main/java/io/ballerina/edi/cmd/CodegenCmd.java]: Changed with https://github.com/ballerina-platform/edi-tools/commit/3c80fe81036dbc049bc518828fc5f8e8aa24d079.

File: edi-cli/src/main/java/io/ballerina/edi/cmd/LibgenCmd.java [hunk 4]
```diff
@@ -21,33 +21,33 @@
...
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 
-@CommandLine.Command(
-        name = "libgen",
-        description = "Generated code for a given EDI schema."
-)
+@CommandLine.Command(name = "libgen", description = "Generated code for a given EDI schema.")
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply edi-cli/src/main/java/io/ballerina/edi/cmd/LibgenCmd.java]: Changed with https://github.com/ballerina-platform/edi-tools/commit/3c80fe81036dbc049bc518828fc5f8e8aa24d079.

File: README.md [hunk 5]
```diff
@@ -1,127 +1,185 @@
...
+ITM*A-45*100~
+ITM*D-10*58~
+ITM*K-80*250~
+ITM*T-46*28~
+```
 
-Ballerina records for the EDI schema in the `edi-schema.json` can be generated as follows (generated Ballerina records will be saved in `orderRecords.bal`):
+## Code generation
+
+The below command can be used to generate records and parsing functions for a given EDI schema:
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply README.md]: Changed with https://github.com/ballerina-platform/edi-tools/commit/3c80fe81036dbc049bc518828fc5f8e8aa24d079.

File: edi-cli/src/main/java/io/ballerina/edi/cmd/EdiCmd.java [hunk 6]
```diff
@@ -21,34 +21,32 @@
...
-        description = "Generate the Ballerina records and parsing functions or a Ballerina package for a given EDI schema or collection of schemas",
-        subcommands = {
-            CodegenCmd.class,
-            LibgenCmd.class,
-            EslCmd.class,
-            ConvertX12Cmd.class,
-            ConvertEdifactCmd.class
-        }
-)
+@CommandLine.Command(name = "edi", description = "Provides the functionality required to process EDI files and implement EDI integrations", subcommands = {
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply edi-cli/src/main/java/io/ballerina/edi/cmd/EdiCmd.java]: All together, the functionality of the **Bal EDI tool** is to process EDI files, which results in giving developers the ability to implement EDI integrations. IMO, 'processing EDI files' includes writing, reading, and/or converting various schemas, and using those schemas for reading and writing `.edi` files.

[the remark this answers is not in the document]
Comment (niveathika) [reply edi-cli/src/main/java/io/ballerina/edi/cmd/EdiCmd.java]: Ack

**Review by niveathika — APPROVED** [review]
LGTM

**Pull request merged**